legion/engine/llri-dx/detail/command_list.cpp

```diff
@@ -0,0 +1,33 @@
+/**
+ * @file command_list.cpp
+ * @copyright 2021-2021 Leon Brands. All rights served.
+ * @license: https://github.com/Legion-Engine/Legion-LLRI/blob/main/LICENSE
+ */
+
+#include <llri/llri.hpp>
+#include <llri-dx/directx.hpp>
+
+namespace LLRI_NAMESPACE
+{
+    result CommandList::impl_begin(const command_list_begin_desc& desc)
+    {
+        //TODO: Handle node mask
+        //TODO: Handle inheritance/indirect
+        const auto r = static_cast<ID3D12GraphicsCommandList*>(m_ptr)->Reset(static_cast<ID3D12CommandAllocator*>(m_groupHandle), nullptr);
+        if (FAILED(r))
+            return directx::mapHRESULT(r);
+
+        m_state = command_list_state::Recording;
+        return result::Success;
```

Comment:
sounds like in DX you can also record without calling reset first? so in this case you might wanna artificially force an error if m_state isn't empty when validation is turned on right?

Reply:
I think the validation checks for states before begin/end already but I can check to verify

Reply:
Yeah, in command_list.inl line 43 I check to make sure m_state is empty so that should be good